### PR TITLE
Add a generic way for creating redirect for dc.org/demo/<name>

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: server/routes/redirects.py [hunk 1]
```diff
@@ -101,4 +106,17 @@ def insights():
...
       url_for('explore.page',
               _scheme=current_app.config.get('SCHEME', 'https'),
-              code=302))
\ No newline at end of file
+              code=302))
+
+
+def load_redirects(name):
+  local_file = gcs.download_file(bucket=GLOBAL_CONFIG_BUCKET,
+                                 filename='redirect.json')
```
Comment (pradh) [inline server/routes/redirects.py:114]: Should we also check-in the file so we have a record of the URLs we have and can restore to GCS, etc.

**Review by pradh — APPROVED** [review]
Very nice!

File: server/routes/redirects.py [hunk 2]
```diff
@@ -101,4 +106,17 @@ def insights():
...
+  with open(local_file) as fp:
+    mapping = json.load(fp)
+    return mapping.get(name, '/')
+
+
+@bp.route('/link/<path:name>')
+def demo(name):
```
Comment (pradh) [inline server/routes/redirects.py:121]: nit: link(name)

**Review by pradh — APPROVED** [review]

File: server/routes/redirects.py (hunk 1 above)
Comment (shifucun) [reply server/routes/redirects.py]: Done

File: server/routes/redirects.py (hunk 2 above)
Comment (shifucun) [reply server/routes/redirects.py]: Done

File: server/routes/redirects.py [hunk 3]
```diff
@@ -101,4 +106,17 @@ def insights():
...
+  local_file = gcs.download_file(bucket=GLOBAL_CONFIG_BUCKET,
+                                 filename='redirects.json')
+  with open(local_file) as fp:
+    mapping = json.load(fp)
+    return mapping.get(name, '/')
+
+
+@bp.route('/link/<path:name>')
+def link(name):
+  return redirect(load_redirects(name), code=302)
```
Comment (keyurva) [inline server/routes/redirects.py:122]: Will this download and load the file from GCS for every link call?

Comment (shifucun) [reply server/routes/redirects.py:122]: Yes, it's doing that to make sure we can dynamically change the content. The performance seems ok.

**Pull request merged**